lib/racecar/daemon.rb

```diff
@@ -0,0 +1,91 @@
+module Racecar
+  class Daemon
+    attr_reader :pidfile
+
+    def initialize(pidfile)
+      @pidfile = pidfile
+    end
+
+    def pidfile?
+      !pidfile.nil?
+    end
+
+    def daemonize!
+      exit if fork
+      Process.setsid
+      exit if fork
+      Dir.chdir "/"
+    end
+
+    def redirect_output(logfile)
+      FileUtils.mkdir_p(File.dirname(logfile), :mode => 0755)
```

Comment:
A hashrocket, such nostalgia. Rubocop would have told you to use `mode: 0755` here. I won’t…